bin/generate-methods.js

```diff
@@ -154,15 +151,39 @@ function generateMethodCalls(node, name) {
     const res = messages.find(method.responseType);
     const loweredName = firstToLower(mname);
 
+    const requestTsType = req.empty ? 'void' : formatType(method.requestType);
+    const responseTsType = res.empty ? 'void' : formatType(method.responseType);
+    const streaming = method.responseStream || method.requestStream;
+
     emit(getCommentPrefixing(`rpc ${mname}(`));
-    emit(`${indent(1)}public ${loweredName}(`
-      + (req.empty ? '' : `req: ${formatType(method.requestType)}`)
-      + '): Promise<' + (res.empty ? 'void' : formatType(method.responseType)) + '> {')
-      (`${indent(2)}return this.client.exec('${name}', '${loweredName}', ${req.empty ? '{}' : 'req'});`)
-      (`${indent(1)}}\n`);
+    emit(`${indent(1)}public ${loweredName}(`);
+    if (!method.requestStream && !req.empty) {
+      emit(`req: ${requestTsType}`);
+    }
+    emit('): ');
+
+    if (method.responseStream && !method.requestStream) {
+      emit(`Promise<IResponseStream<${responseTsType}>> {\n`)
+        (`${indent(2)}return this.client.getConnection('${name}').then(cnx => {\n`)
+        (`${indent(3)}return cnx.${loweredName}(${req.empty ? '{}' : 'req'});\n`)
+        (`${indent(2)}});\n`)
+    } else if (method.responseStream && method.requestStream) {
+      emit(`Promise<IDuplexStream<${requestTsType}, ${responseTsType}>> {\n`)
+        (`${indent(2)}return this.client.getConnection('${name}').then(cnx => {\n`)
```

Comment:
That would lead to weird indentation. I might use lodash templates though so that they can be statically defined without all this mess